core/trino-spi/src/main/java/io/trino/spi/statistics/StatsUtil.java

```diff
@@ -85,4 +85,55 @@ public static OptionalDouble toStatsRepresentation(Type type, Object value)
 
         return OptionalDouble.empty();
     }
+
+    /**
+     * Stats representations of the given types are considered compatible if
+     * {@link #toStatsRepresentation} returns the same result for all valid values with either type.
```

Comment:
This isn't precise. Do you mean values that can be cast from one type to the other and back, without loss?

Reply:
Also, we should be clear whether we're defining an equivalence relation.

- Is this reflective? yes
- is this symmetric? currently it is (per how the method is defined & named; but maybe we want to model this around wider/narrower types and then it wouldn't be)
- is this transitive?